frontend/src/pages/QueryBuilder/QueryBuilder.tsx

```diff
@@ -127,30 +142,57 @@ class QueryBuilder extends React.Component<QueryBuilderProps> {
     const editable = this.isEditable(operation);
 
     return (
-      <OperationForm
-        operation={operation}
-        editable={editable}
-        valid={steps.count() > 0}
-        onUpdateOperation={this.onUpdateOperation}
-        onDuplicateOperation={this.onDuplicateOperation}
-        onSuccess={this.onSaveOperation}
-        processing={this.props.page.get('processing') as boolean}
-        onDeleteOperation={this.onDeleteOperation}
-        onReset={!id ? () => this.props.actions.setActiveOperation() : undefined}
-      >
-        <OperationSteps
-          sources={this.props.sources.get('sources') as List<SourceMap>}
-          isFetchingSources={this.props.sources.get('loading') as boolean}
-          steps={steps}
-          fetchSources={this.props.actions.fetchSources}
-          onSelectSource={this.props.actions.setActiveSource}
-          onAddStep={this.props.actions.updateActiveStep}
-          activeSource={this.props.activeSource}
-          activeStep={activeStep}
-          onClickStep={(step) => this.props.actions.updateActiveStep(step, true)}
+      <>
+        <BasicModal
+          show={this.state.modalShow}
+          onHide={() => {
+            this.setState((state) => {
+              return { modalShow: !state.modalShow };
+            });
+          }}
+        >
+          <p>{this.renderTable()}</p>
+        </BasicModal>
```

Comment:
I'd rather we don't have this as a modal ... I'd rather the preview gets shown in the same place as the step form... should be an easy addition into this component. I imagine if there's an active step, u can still preview & then return to the active step once the preview is closed.

Reply:
Updated